src/vma/dev/cq_mgr.cpp

```diff
@@ -694,24 +694,23 @@ void cq_mgr::reclaim_recv_buffer_helper(mem_buf_desc_t* buff)
 				}
 			#endif
 				temp = buff;
-				buff = temp->p_next_desc;
-				temp->p_next_desc = NULL;
-				temp->p_prev_desc = NULL;
-				temp->reset_ref_count();
-				temp->rx.tcp.gro = 0;
-				temp->rx.is_vma_thr = false;
-#ifdef DEFINED_VMAPOLL				
-				temp->rx.vma_polled = false;
-#endif // DEFINED_VMAPOLL				
+				free_lwip_pbuf(&temp->lwip_pbuf);
+				temp->rx.hw_raw_timestamp = 0;
 				temp->rx.flow_tag_id = 0;
-				temp->rx.tcp.p_ip_h = NULL;
-				temp->rx.tcp.p_tcp_h = NULL;
+
 				temp->rx.udp.sw_timestamp.tv_nsec = 0;
 				temp->rx.udp.sw_timestamp.tv_sec = 0;
 				temp->rx.udp.hw_timestamp.tv_nsec = 0;
 				temp->rx.udp.hw_timestamp.tv_sec = 0;
-				temp->rx.hw_raw_timestamp = 0;
-				free_lwip_pbuf(&temp->lwip_pbuf);
+
```

Comment:
Unfortunately moving clean up code to specific method negatively impacted performance even with forcing inline, so it stayed at origin places.